sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -40,6 +40,13 @@ static void _rpl_trickle_send_dio(void *args)
 {
     gnrc_rpl_dodag_t *dodag = (gnrc_rpl_dodag_t *) args;
     ipv6_addr_t all_RPL_nodes = GNRC_RPL_ALL_NODES_ADDR;
+
+    /* a leaf node does not send DIOs periodically */
```

Comment:
Could you elaborate on `occasionally`? IMO, when a node becomes a leaf and announces this by sending an `INFINITE_RANK`, then there is no reason anymore to send out occasional `DIOs`. If a neighbor does not receive this `DIO` and has this leaf as a parent, then the lifetime will not be refreshed and thus a new parent will be selected after the timeout.